addons/nuxeo-core-binarymanager-cloud/nuxeo-core-binarymanager-s3/src/main/java/org/nuxeo/ecm/blob/s3/S3BlobStoreConfiguration.java

```diff
@@ -278,8 +279,14 @@ public S3BlobStoreConfiguration(Map<String, String> properties) throws IOExcepti
         amazonS3 = getAmazonS3(s3Builder);
 
         metadataAddUsername = getBooleanProperty(METADATA_ADD_USERNAME_PROPERTY);
-        bucketRetentionMode = computeBucketRetentionMode();
-        retentionMode = bucketRetentionMode == null ? DEFAULT_RETENTION_MODE : bucketRetentionMode;
+
+        s3RetentionEnabled = isS3RetentionEnabled();
+        retentionMode = Framework.isBooleanPropertyTrue(PROP_RETENTION_COMPLIANCE_MODE_ENABLED)
+                ? ObjectLockRetentionMode.COMPLIANCE
+                : ObjectLockRetentionMode.GOVERNANCE;
+        if (!s3RetentionEnabled && Boolean.parseBoolean(properties.get(RECORD))) {
+            log.warn("Bucket {} is tagged for records but retention is not enabled at s3 level", bucketName);
```

Comment:
"is tagged for" is a strange turn of phrase. I would say "is configured for". Also it's not the bucket that is configured for records, it's the blob provider.
Also s3 -> S3.